$OpenBSD$

https://gitlab.haskell.org/ghc/ghc/-/merge_requests/3924

[fixup 3433] move debugBelch into IF_DEBUG(linker)

The commit in dff1cb3d9c111808fec60190747272b973547c52 incorrectly left
the `debugBelch` function without a comment or IF_DEBUG(linker,)
decoration. This rectifies it.

Needs at least a 8.10 backport, as it was backported in 6471cc6aff80d5deebbdb1bf7b677b31ed2af3d5

Index: rts/linker/Elf.c
--- rts/linker/Elf.c.orig
+++ rts/linker/Elf.c
@@ -903,8 +903,9 @@ ocGetNames_ELF ( ObjectCode* oc )
                    common_used += symbol->elf_sym->st_size;
                    ASSERT(common_used <= common_size);
 
-                   debugBelch("COMMON symbol, size %ld name %s allocated at %p\n",
-                                   symbol->elf_sym->st_size, nm, symbol->addr);
+                   IF_DEBUG(linker,
+                            debugBelch("COMMON symbol, size %ld name %s allocated at %p\n",
+                                       symbol->elf_sym->st_size, nm, symbol->addr));
 
                    /* Pointless to do addProddableBlock() for this area,
                       since the linker should never poke around in it. */
